src/eduid_webapp/group_management/views/invite.py

```diff
@@ -0,0 +1,209 @@
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided
+#        with the distribution.
+#     3. Neither the name of the NORDUnet nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+from typing import Mapping
+from uuid import UUID
+
+from flask import Blueprint
+from pymongo.errors import DuplicateKeyError
+
+from eduid_common.api.decorators import MarshalWith, UnmarshalWith, require_user
+from eduid_common.api.exceptions import MailTaskFailed
+from eduid_common.api.messages import CommonMsg, error_message
+from eduid_groupdb import User as GraphUser
+from eduid_scimapi.userdb import ScimApiUser
+from eduid_userdb import User
+from eduid_userdb.exceptions import DocumentDoesNotExist, EduIDDBError
+from eduid_userdb.group_management import GroupInviteState
+
+from eduid_webapp.group_management.app import current_group_management_app as current_app
+from eduid_webapp.group_management.helpers import (
+    GroupManagementMsg,
+    accept_group_invitation,
+    get_incoming_invites,
+    get_outgoing_invites,
+    get_scim_user_by_eppn,
+    is_owner,
+    send_invite_email,
+)
+from eduid_webapp.group_management.schemas import (
+    GroupAllInviteResponseSchema,
+    GroupIncomingInviteResponseSchema,
+    GroupInviteRequestSchema,
+    GroupOutgoingInviteResponseSchema,
+)
+
+__author__ = 'lundberg'
+
+group_invite_views = Blueprint('group_invite', __name__, url_prefix='/invites/', template_folder='templates')
+
+
+@group_invite_views.route('/all', methods=['GET'])
+@MarshalWith(GroupAllInviteResponseSchema)
+@require_user
+def all_invites(user: User) -> Mapping:
+    incoming = get_incoming_invites(user)
+    outgoing = []
+    scim_user = get_scim_user_by_eppn(user.eppn)
+    if scim_user:
+        # The user can only be a group owner if there is a scim_user
+        graph_user = GraphUser(identifier=str(scim_user.scim_id))
+        owner_groups = current_app.scimapi_groupdb.get_groups_for_owner(owner=graph_user)
```

Comment:
it feels like we are creating these GraphUser instances with only the identifier set a lot? Maybe the API is wrong and the functions like get_groups_for_owner should accept identifiers instead of GraphUsers?

Reply:
Yes